core/Azure.Mcp.Core/src/Areas/Tools/Commands/ToolsListCommand.cs

```diff
@@ -97,13 +112,49 @@ public override async Task<CommandResponse> ExecuteAsync(CommandContext context,
                     }
                 }
 
+                // If --name-only is also specified, return only the names
+                if (options.NameOnly)
+                {
+                    var namespaceNames = namespaceCommands.Select(nc => nc.Command).ToList();
+                    var result = new ToolNamesResult(namespaceNames);
+                    context.Response.Results = ResponseResult.Create(result, ModelsJsonContext.Default.ToolNamesResult);
+                    return context.Response;
+                }
+
                 context.Response.Results = ResponseResult.Create(namespaceCommands, ModelsJsonContext.Default.ListCommandInfo);
                 return context.Response;
             }
 
-            var tools = await Task.Run(() => CommandFactory.GetVisibleCommands(factory.AllCommands)
-                .Select(kvp => CreateCommand(kvp.Key, kvp.Value))
-                .ToList());
+            // If the --name-only flag is set (without namespace mode), return only tool names
+            if (options.NameOnly)
+            {
+                // Get all visible commands and extract their tokenized names (full command paths)
+                var allToolNames = CommandFactory.GetVisibleCommands(factory.AllCommands)
+                    .Select(kvp => kvp.Key) // Use the tokenized key instead of just the command name
+                    .Where(name => !string.IsNullOrEmpty(name));
+
+                // Apply namespace filtering if specified (using underscore separator for tokenized names)
+                allToolNames = ApplyNamespaceFilterToNames(allToolNames, options.Namespaces, '_');
```

Comment:
We could use the `CommandFactory.Separator` instead of hardcoding the underscore char.